### ISSUE TITLE
Filename too long error message - how to work it around?

### ISSUE THREAD

**Issue by Saboteur777** [spacecatninja/craft-imager-x#194]
### I'm submitting a... 

- [ ] bug report
- [ ] feature request
- [x] general question

### Steps to reproduce

1.Upload a file called `Thomas20Bach20a20Nemzetközi20Olimpiai20Bizottság20a20NOB20elnöke20k20Csoj20Mun20Szun20a20dél-koreai20Kangvon20tartomány20kormányzója20b20és20Pak20Jang20Vu20dél-koreai20kulturális20sport20és20turisztikai20miniszter87_58ecec703ef264d6784fae00a42ba683.jpg` to `/var/www/web/imager/photos/`
2. Try to generate transform for it with `craft imager-x/generate/index -v photos`
3. Error message is thrown: 

```
 > Volume `photos`
> Not recursive
> Generating 2 named transforms for 27394 images.
6% (3324/54788) ETA: 320 sec.   Error: Save operation failed
2023-01-18 23:22:20 [console.ERROR] [Imagine\Exception\RuntimeException] Save operation failed {"trace":["#0 /var/www/vendor/spacecatninja/imager-x/src/transformers/CraftTransformer.php(283): Imagine\\Imagick\\Image->save()","#1 /var/www/vendor/spacecatninja/imager-x/src/transformers/CraftTransformer.php(97): spacecatninja\\imagerx\\transformers\\CraftTransformer->getTransformedImage()","#2 /var/www/vendor/spacecatninja/imager-x/src/services/ImagerService.php(477): spacecatninja\\imagerx\\transformers\\CraftTransformer->transform()","#3 /var/www/vendor/spacecatninja/imager-x/src/services/GenerateService.php(294): spacecatninja\\imagerx\\services\\ImagerService->transformImage()","#4 /var/www/vendor/spacecatninja/imager-x/src/console/controllers/GenerateController.php(193): spacecatninja\\imagerx\\services\\GenerateService->generateTransformsForAsset()","#5 [internal function]: spacecatninja\\imagerx\\console\\controllers\\GenerateController->actionIndex()","#6 /var/www/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array()","#7 /var/www/vendor/yiisoft/yii2/base/Controller.php(178): yii\\base\\InlineAction->runWithParams()","#8 /var/www/vendor/yiisoft/yii2/console/Controller.php(180): yii\\base\\Controller->runAction()","#9 /var/www/vendor/yiisoft/yii2/base/Module.php(552): yii\\console\\Controller->runAction()","#10 /var/www/vendor/yiisoft/yii2/console/Application.php(180): yii\\base\\Module->runAction()","#11 /var/www/vendor/craftcms/cms/src/console/Application.php(90): yii\\console\\Application->runAction()","#12 /var/www/vendor/yiisoft/yii2/console/Application.php(147): craft\\console\\Application->runAction()","#13 /var/www/vendor/craftcms/cms/src/console/Application.php(121): yii\\console\\Application->handleRequest()","#14 /var/www/vendor/yiisoft/yii2/base/Application.php(384): craft\\console\\Application->handleRequest()","#15 /var/www/craft(13): yii\\base\\Application->run()","#16 {main}"],"memory":221010336,"exception":"[object] (Imagine\\Exception\\RuntimeException(code: 435): Save operation failed at /var/www/vendor/pixelandtonic/imagine/src/Imagick/Image.php:385)\n[previous exception] [object] (ImagickException(code: 435): unable to open image '/var/www/web/imager/photos/324344/Thomas20Bach20a20Nemzetközi20Olimpiai20Bizottság20a20NOB20elnöke20k20Csoj20Mun20Szun20a20dél-koreai20Kangvon20tartomány20kormányzója20b20és20Pak20Jang20Vu20dél-koreai20kulturális20sport20és20turisztikai20miniszter87_58ecec703ef264d6784fae00a42ba683.jpg': Filename too long @ error/blob.c/OpenBlob/3570 at /var/www/vendor/pixelandtonic/imagine/src/Imagick/Image.php:383)"} 

```

### Description
How could I prevent this to happen? I am migrating a pretty big site and it can happen that there are multiple files with too long filenames. 

### Displayed errors, stack trace, relevant logs  

```
 > Volume `photos`
> Not recursive
> Generating 2 named transforms for 27394 images.
6% (3324/54788) ETA: 320 sec.   Error: Save operation failed
2023-01-18 23:22:20 [console.ERROR] [Imagine\Exception\RuntimeException] Save operation failed {"trace":["#0 /var/www/vendor/spacecatninja/imager-x/src/transformers/CraftTransformer.php(283): Imagine\\Imagick\\Image->save()","#1 /var/www/vendor/spacecatninja/imager-x/src/transformers/CraftTransformer.php(97): spacecatninja\\imagerx\\transformers\\CraftTransformer->getTransformedImage()","#2 /var/www/vendor/spacecatninja/imager-x/src/services/ImagerService.php(477): spacecatninja\\imagerx\\transformers\\CraftTransformer->transform()","#3 /var/www/vendor/spacecatninja/imager-x/src/services/GenerateService.php(294): spacecatninja\\imagerx\\services\\ImagerService->transformImage()","#4 /var/www/vendor/spacecatninja/imager-x/src/console/controllers/GenerateController.php(193): spacecatninja\\imagerx\\services\\GenerateService->generateTransformsForAsset()","#5 [internal function]: spacecatninja\\imagerx\\console\\controllers\\GenerateController->actionIndex()","#6 /var/www/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array()","#7 /var/www/vendor/yiisoft/yii2/base/Controller.php(178): yii\\base\\InlineAction->runWithParams()","#8 /var/www/vendor/yiisoft/yii2/console/Controller.php(180): yii\\base\\Controller->runAction()","#9 /var/www/vendor/yiisoft/yii2/base/Module.php(552): yii\\console\\Controller->runAction()","#10 /var/www/vendor/yiisoft/yii2/console/Application.php(180): yii\\base\\Module->runAction()","#11 /var/www/vendor/craftcms/cms/src/console/Application.php(90): yii\\console\\Application->runAction()","#12 /var/www/vendor/yiisoft/yii2/console/Application.php(147): craft\\console\\Application->runAction()","#13 /var/www/vendor/craftcms/cms/src/console/Application.php(121): yii\\console\\Application->handleRequest()","#14 /var/www/vendor/yiisoft/yii2/base/Application.php(384): craft\\console\\Application->handleRequest()","#15 /var/www/craft(13): yii\\base\\Application->run()","#16 {main}"],"memory":221010336,"exception":"[object] (Imagine\\Exception\\RuntimeException(code: 435): Save operation failed at /var/www/vendor/pixelandtonic/imagine/src/Imagick/Image.php:385)\n[previous exception] [object] (ImagickException(code: 435): unable to open image '/var/www/web/imager/photos/324344/Thomas20Bach20a20Nemzetközi20Olimpiai20Bizottság20a20NOB20elnöke20k20Csoj20Mun20Szun20a20dél-koreai20Kangvon20tartomány20kormányzója20b20és20Pak20Jang20Vu20dél-koreai20kulturális20sport20és20turisztikai20miniszter87_58ecec703ef264d6784fae00a42ba683.jpg': Filename too long @ error/blob.c/OpenBlob/3570 at /var/www/vendor/pixelandtonic/imagine/src/Imagick/Image.php:383)"} 

```

### Additional info

- Imager version & edition: 4.1.8
- Craft version: Craft Pro 4.3.6.1
- PHP version: 8.1.14
- Image driver & version: Imagick 3.7.0 (ImageMagick 7.1.0-52)

**Comment by aelvan** [comment]
Hi,

I'll try to come up with a solution that address this. I think I'll have to implement some sort of character limit on filenames, just not 100% sure what would be a good solution for this when considering backwards compatibility and side-effects.

By the way, what OS is this?

**Comment by Saboteur777** [comment]
Thanks! It happened on a Linux VPS, but the file came from an FTP server. The file was saved by a custom CMS (yay... -.-), so that explains the weird filename. 

I solved it by renaming the file. The situation would have been better if the pre-transform could just skip this file instead of blocking the process completely.

**Comment by aelvan** [comment]
Fixed in 4.1.10.
